### PR TITLE
Initial version of the CurieAudio library.

### PR DESCRIPTION
**Pull request by SidLeung**
This library originates from the CurieI2S\src\CurieI2SDMA module.  It forms the physical layer of the Sound library concept from Arduio.
The library utilizes two drivers:  soc_i2s and soc_dma.  The library performs I2S I/O in blocks of audio data via DMA cbannels.

### CODE REVIEW

**Comment by SidLeung** [comment]
This PR is created to generate JSON.  Code is not up for review yet.

File: libraries/CurieAudio/src/I2SController.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if(enable)
+    {
+        mux_mode = I2S_MUX_MODE;
+    }
+    
+    /* Set SoC pin mux configuration */
+    SET_PIN_MODE(g_APinDescription[I2S_TXD].ulSocPin, mux_mode);
+    SET_PIN_MODE(g_APinDescription[I2S_TWS].ulSocPin, mux_mode);
+    SET_PIN_MODE(g_APinDescription[I2S_TSCK].ulSocPin, mux_mode);
+    g_APinDescription[I2S_TXD].ulPinMode = mux_mode;
```
Comment (eriknyquist) [inline libraries/CurieAudio/src/I2SController.cpp]: These lines (53-55) will throw a compile error after https://github.com/01org/corelibs-arduino101/pull/308 is merged
(same goes for any further modifications to g_APinDescription).

The above pull request has already undergone review & testing-- I suggest you review it yourself, and rebase once it is merged (which will probably be today).

File: libraries/CurieAudio/src/I2SController.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  I2S_MISC_ERROR
+} i2sErrorCode;
+
+
+class I2SController
+{
+    public:
+        I2SController();
+
+        //
```
Comment (eriknyquist) [inline libraries/CurieAudio/src/I2SController.h]: Empty commented lines here on 87, 93

File: system/libarc32_arduino101/drivers/soc_i2s.c [hunk 3]
```diff
@@ -228,7 +229,7 @@ static void i2s_dma_cb_block(void *num)
 DRIVER_API_RC soc_i2s_config(uint8_t channel, struct soc_i2s_cfg *cfg)
 {
 	uint32_t reg;
-	uint16_t sample_rate;
+	//	uint16_t sample_rate;
```
Comment (eriknyquist) [inline system/libarc32_arduino101/drivers/soc_i2s.c]: If there is a good reason for these commented lines (232, 251), can you add more comments to explain when/why they might be needed?

**Pull request closed without merging**